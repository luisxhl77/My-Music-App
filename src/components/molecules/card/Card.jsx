import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { PlayCircleFilled } from '@mui/icons-material';
import './card.scss';

export const Card = ({ name, description, image, type, id }) => {

    return (
        <div className="card">
            <Link to={ `/playlist/${id}` } className="card__textLink"> 
                <div className="card__play-music">
                    <div className="card__image">
                        <img src={ image } alt="cover" className="card__img" id="disk1" aria-label='imageCard'/>
                    </div>
                    <PlayCircleFilled fontSize="large" className="btnPlay"/>
                </div>
                <h5 className="card__name-playlist" aria-label='nameCard'>{ name }</h5>
                <p className="card__description-playlist" aria-label='descriptionCard'>{ description }</p>
                <p className="card__description-playlist" aria-label='typeCard'>{ type }</p>
            </Link>
        </div>
    )
}

Card.propTypes = {
    id          : PropTypes.string.isRequired,
    name        : PropTypes.string.isRequired,
    type        : PropTypes.string,
    image       : PropTypes.string,
    description : PropTypes.string.isRequired,
}

Card.defaultProps = {
    name        : 'no title',
    type        : 'no type',
    image       : '../../../img/iconSpotify.jpg',
    description : 'No description',
}