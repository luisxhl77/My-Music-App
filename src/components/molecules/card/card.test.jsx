import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Card } from "./Card";

describe('Pruebas de <Card/>', () => {

    const id          = 'ID_2174123';
    const type        = 'Playlist';
    const name        = 'This is Avicii';
    const image       = 'https://avicii.com/avicii.jsg';
    const description = 'Avicii';
    
    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render(
            <MemoryRouter>
                <Card
                    id          = { id }
                    type        = { type }
                    name        = { name }
                    image       = { image }
                    description = { description }
                /> 
            </MemoryRouter>
        );
        
        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar src de la imagen', () => {
   
        render(
            <MemoryRouter>
                <Card
                    id          = { id }
                    type        = { type }
                    name        = { name }
                    image       = { image }
                    description = { description }
                /> 
            </MemoryRouter>
        );

        const { src } = screen.getByLabelText('imageCard');
        expect( src ).toBe( image );

    })

    test('debe de mostrar el nombre el tipo y la descripcion de la card', () => {
   
        render(
            <MemoryRouter>
                <Card
                    id          = { id }
                    name        = { name }
                    type        = { type }
                    image       = { image }
                    description = { description }
                /> 
            </MemoryRouter>
        );

        expect( screen.getByText(    name     ) ).toBe( screen.getByLabelText(     'nameCard'    ) );
        expect( screen.getByText(    type     ) ).toBe( screen.getByLabelText(     'typeCard'    ) );
        expect( screen.getByText( description ) ).toBe( screen.getByLabelText( 'descriptionCard' ) );

    })

})