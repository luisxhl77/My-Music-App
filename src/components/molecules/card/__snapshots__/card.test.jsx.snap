// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <Card/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <a
      class="card__textLink"
      href="/playlist/ID_2174123"
    >
      <div
        class="card__play-music"
      >
        <div
          class="card__image"
        >
          <img
            alt="cover"
            aria-label="imageCard"
            class="card__img"
            id="disk1"
            src="https://avicii.com/avicii.jsg"
          />
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge btnPlay css-tzssek-MuiSvgIcon-root"
          data-testid="PlayCircleFilledIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 14.5v-9l6 4.5-6 4.5z"
          />
        </svg>
      </div>
      <h5
        aria-label="nameCard"
        class="card__name-playlist"
      >
        This is Avicii
      </h5>
      <p
        aria-label="descriptionCard"
        class="card__description-playlist"
      >
        Avicii
      </p>
      <p
        aria-label="typeCard"
        class="card__description-playlist"
      >
        Playlist
      </p>
    </a>
  </div>
</div>
`;
